sdk/vppagent/flush.go

```diff
@@ -0,0 +1,172 @@
+		return nil, err
+	}
+
+	return incomingConnection, nil
+}
+
+// Close implements the close handler
+func (f *Flush) Close(ctx context.Context, connection *connection.Connection) (*empty.Empty, error) {
+	opaque := f.GetNext().GetOpaque(connection)
+	if opaque != nil {
+		dataChange := opaque.(*ConnectionData).DataChange
+
+		logrus.Infof("Removing DataChange from VPP Agent: %v", dataChange)
+		err := f.remove(ctx, dataChange)
+		if err != nil {
+			logrus.Errorf("Failed to remove DataChange from VPP Agent: %v", err)
+			return &empty.Empty{}, err
+		}
+	}
+
+	err := f.Conn.Close()
+	if err != nil {
+		logrus.Errorf("Failed to close connection to VPP Agent: %v", err)
+		return &empty.Empty{}, err
+	}
+
+	if f.GetNext() != nil {
+		return f.GetNext().Close(ctx, connection)
+	}
+	return &empty.Empty{}, nil
+}
+
+// Name returns the composite name
+func (f *Flush) Name() string {
+	return "flush"
+}
+
+// NewFlush creates a Flush
+func NewFlush(configuration *common.NSConfiguration, endpoint string) *Flush {
+	// ensure the env variables are processed
+	if configuration == nil {
+		configuration = &common.NSConfiguration{}
+	}
+	configuration.CompleteNSConfiguration()
+
+	conn, err := createConnection(endpoint)
+	if err != nil {
+		logrus.Errorf("Failed to create connection to VPP Agent: %v", err)
+		return nil
+	}
+
+	self := &Flush{
+		Conn: conn,
+	}
+
+	logrus.Infof("Resetting VPP Agent")
+	err = self.reset()
+	if err != nil {
+		logrus.Errorf("Failed to reset VPP Agent: %v", err)
+		return nil
+	}
+
+	return self
+}
+
+func createConnection(endpoint string) (*grpc.ClientConn, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	if err := tools.WaitForPortAvailable(ctx, "tcp", endpoint, 100*time.Millisecond); err != nil {
+		return nil, err
+	}
+
+	tracer := opentracing.GlobalTracer()
+	rv, err := grpc.Dial(endpoint, grpc.WithInsecure(),
+		grpc.WithUnaryInterceptor(
+			otgrpc.OpenTracingClientInterceptor(tracer, otgrpc.LogPayloads())),
+		grpc.WithStreamInterceptor(
+			otgrpc.OpenTracingStreamClientInterceptor(tracer)))
+
+	if err != nil {
+		logrus.Errorf("Can't dial grpc server: %v", err)
+		return nil, err
+	}
+
+	return rv, nil
+}
+
+func (f *Flush) send(ctx context.Context, dataChange *configurator.Config) error {
+	client := configurator.NewConfiguratorClient(f.Conn)
+
+	if _, err := client.Update(ctx, &configurator.UpdateRequest{Update: dataChange}); err != nil {
+		logrus.Error(err)
+		_, _ = client.Delete(ctx, &configurator.DeleteRequest{Delete: dataChange})
+		return err
+	}
+	return nil
+}
+
+func (f *Flush) remove(ctx context.Context, dataChange *configurator.Config) error {
+	client := configurator.NewConfiguratorClient(f.Conn)
+
+	if _, err := client.Delete(ctx, &configurator.DeleteRequest{Delete: dataChange}); err != nil {
+		logrus.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (f *Flush) reset() error {
```

Comment:
Why it is named reset?

Reply:
1) It was named like that before:
https://github.com/networkservicemesh/networkservicemesh/blob/master/test/applications/cmd/vppagent-icmp-responder-nse/vppagent.go#L56
https://github.com/networkservicemesh/networkservicemesh/blob/c3e247e9a672330247b13dcbfe3eb416e2689a9b/test/applications/cmd/vppagent-firewall-nse/vppagent.go#L77

2) It resets VPP Agent config